java/rocksjni/ratelimiterjni.cc

```diff
@@ -22,3 +23,98 @@ jlong Java_org_rocksdb_GenericRateLimiterConfig_newRateLimiterHandle(
       static_cast<int64_t>(jrefill_period_micros),
       static_cast<int32_t>(jfairness)));
 }
+
+/*
+ * Class:     org_rocksdb_RateLimiter
+ * Method:    newRateLimiterHandle
+ * Signature: (JJI)J
+ */
+jlong Java_org_rocksdb_RateLimiter_newRateLimiterHandle(
+    JNIEnv* env, jclass jclazz, jlong jrate_bytes_per_second,
+    jlong jrefill_period_micros, jint jfairness) {
+  auto* rate_limiter = rocksdb::NewGenericRateLimiter(
+      static_cast<int64_t>(jrate_bytes_per_second),
+      static_cast<int64_t>(jrefill_period_micros),
+      static_cast<int32_t>(jfairness));
+
+  // TODO(AR) `ptr_sptr_rate_limiter` is never deleted :-(
+  std::shared_ptr<rocksdb::RateLimiter> *ptr_sptr_rate_limiter =
+    new std::shared_ptr<rocksdb::RateLimiter>;
+  *ptr_sptr_rate_limiter = std::shared_ptr<rocksdb::RateLimiter>(rate_limiter);
+
+  return reinterpret_cast<jlong>(ptr_sptr_rate_limiter);
+}
+
+/*
+ * Class:     org_rocksdb_RateLimiter
+ * Method:    disposeInternal
+ * Signature: (J)V
+ */
+void Java_org_rocksdb_RateLimiter_disposeInternal(
+    JNIEnv* env, jobject jobj, jlong jhandle) {
+  // TODO(AR) `ptr_sptr_rate_limiter` is never deleted :-(
+  std::shared_ptr<rocksdb::RateLimiter> *handle =
+      reinterpret_cast<std::shared_ptr<rocksdb::RateLimiter> *>(jhandle);
+  handle->reset();
```

Comment:
Yes, multiple objects could share same RateLimiter, but the ownership of RateLimiter is already taken care of by the shared_ptr itself (i.e., by calling handle->reset().)  Deleting `handle` here means we delete the pointer to the shared_ptr (i.e., nothing related to the object pointed by the shared_ptr), meaning the Java object that is about to go out of scope can no longer store other shared_ptr of RateLimiter.

Am I right or wrong at this part?